src/PostgREST/Config.hs

```diff
@@ -222,12 +222,21 @@ readAppConfig dbSettings optPath prevDbUri roleSettings roleIsolationLvl = do
     Left err ->
       return . Left $ "Error in config " <> err
     Right parsedConfig ->
-      Right <$> decodeLoadFiles parsedConfig
+      case processForbiddenValues parsedConfig of
+        Left err' ->
+          return . Left $ "Error in config " <> err'
+        Right config ->
+          Right <$> decodeLoadFiles config
   where
     -- Both C.ParseError and IOError are shown here
     loadConfig :: FilePath -> IO (Either SomeException C.Config)
     loadConfig = try . C.load
 
+    processForbiddenValues :: AppConfig -> Either Text AppConfig
+    processForbiddenValues cfg@AppConfig{..}
+      | configAdminServerPort == Just configServerPort = Left "admin-server-port cannot be the same as server-port"
+      | otherwise = Right cfg
```

Comment:
Could this be done in the `parser`? You could create and reuse a `parseServerPort "server-port"`  function and do the validation when parsing the `admin-server-port`. Something like (an idea, may have errors):

```haskell
 parseAdminServerPort k = do
      serverPort <- parseServerPort "server-port"
      optString k >>= \case
        Nothing -> pure Nothing
        Just asp | asp == serverPort -> fail "admin-server-port cannot be the same as server-port"
                 | otherwise -> pure $ Just asp
```

Reply:
Done!